openshift/buildConfig-template.yaml

```diff
@@ -0,0 +1,152 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: graph-backup-buildconfig
+  annotations:
+    description: >
+      This is Thoth Graph Backup CronJob BuildConfig, this template is meant to be used by Bots,
+      but could also be used by humans...
+    openshift.io/display-name: "Thoth: Graph Backup Job BuildConfig"
+    version: 0.1.0
+    tags: thoth,graph-backup,ai-stacks,aistacks,graph
+    template.openshift.io/documentation-url: https://github.com/Thoth-Station/
+    template.openshift.io/long-description: >
+      This template defines resources needed to deploy Thoth Graph Backup Job BuildConfig to OpenShift.
+    template.openshift.io/provider-display-name: Red Hat, Inc.
+    thoth-station.ninja/template-version: 0.1.0
+  labels:
+    template: graph-backup-buildconfig
+    app: thoth
+    component: configmap
+
+objects:
+  - kind: BuildConfig
+    apiVersion: build.openshift.io/v1
+    metadata:
+      name: s2i-thoth-ubi8-py36-postgres
+      annotations:
+        thoth-station.ninja/template-version: 0.1.0
+      labels:
+        app: thoth
+        component: graph-backup
+    spec:
+      resources:
+        requests:
+          memory: "256Mi"
+          cpu: "500m"
+        limits:
+          memory: "256Mi"
+          cpu: "500m"
+      output:
+        to:
+          kind: ImageStreamTag
+          name: "s2i-thoth-ubi8-py36-postgres:${IMAGE_STREAM_TAG}"
+      source:
+        dockerfile: |
+          FROM s2i-thoth-ubi8-py36:latest
+          ENV LANG=en_US.UTF-8
+          USER 0
+          RUN yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm
+          RUN yum install -y postgresql
```

Comment:
I think we need to match with the version running in the cluster - `postgresql96` (I think encountered some failures with when having different versions locally). Also, there is no postgres rpm in ubi? :thinking:

Reply:
@fridex `postgresql96` was failing in openshift, i was testing this in production env. 
```
warning: /var/cache/dnf/pgdg96-b6cbc5f1e3bd22e1/packages/postgresql96-9.6.16-1PGDG.rhel8.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 442df0f8: NOKEY
PostgreSQL 9.6 for RHEL/CentOS 8 - x86_64       1.6 MB/s | 1.7 kB     00:00    
Importing GPG key 0x442DF0F8:
 Userid     : "PostgreSQL RPM Building Project <pgsqlrpms-hackers@pgfoundry.org>"
 Fingerprint: 68C9 E2B9 1A37 D136 FE74 D176 1F16 D2E1 442D F0F8
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64            1/2 
  Verifying        : postgresql96-9.6.16-1PGDG.rhel8.x86_64                 1/2 
  Verifying        : postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64            2/2 
Installed products updated.

Failed:
  postgresql96-9.6.16-1PGDG.rhel8.x86_64                                        
  postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64                                   

error: rpmdb: BDB0060 PANIC: fatal region error detected; run recovery
error: db5 error(-30973) from db->close: BDB0087 DB_RUNRECOVERY: Fatal error, run database recovery
...
error: rpmdb: BDB1582 Open file handle: /var/lib/rpm/__db.001
...
error: rpmdb: BDB0060 PANIC: fatal region error detected; run recovery
error: db5 error(-30973) from dbenv->close: BDB0087 DB_RUNRECOVERY: Fatal error, run database recovery
Error: Transaction failed
``` 
In `thoth-test-core` it failed abruptly:
```

Subscription Manager is operating in container mode.
--
  | PostgreSQL 12 for RHEL/CentOS 8 - x86_64        350 kB/s \| 586 kB     00:01
  | PostgreSQL 11 for RHEL/CentOS 8 - x86_64        486 kB/s \| 698 kB     00:01
  | PostgreSQL 10 for RHEL/CentOS 8 - x86_64        922 kB/s \| 597 kB     00:00
  | PostgreSQL 9.6 for RHEL/CentOS 8 - x86_64       645 kB/s \| 576 kB     00:00
  | PostgreSQL 9.5 for RHEL/CentOS 8 - x86_64       297 kB/s \| 483 kB     00:01
  | PostgreSQL 9.4 for RHEL/CentOS 8 - x86_64       436 kB/s \| 415 kB     00:00
  | Red Hat Enterprise Linux 8 for x86_64 - AppStre  53 kB/s \| 2.8 kB     00:00
  | Red Hat Enterprise Linux 8 for x86_64 - BaseOS   45 kB/s \| 2.3 kB     00:00
  | No match for argument: postgresql96
  | Error: Unable to find a match: postgresql96
``` 
In local docker build `postgresql96` has no issues. 
```
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
PostgreSQL 12 for RHEL/CentOS 8 - x86_64        855 kB/s | 586 kB     00:00    
PostgreSQL 11 for RHEL/CentOS 8 - x86_64        1.4 MB/s | 698 kB     00:00    
PostgreSQL 10 for RHEL/CentOS 8 - x86_64        464 kB/s | 597 kB     00:01    
PostgreSQL 9.6 for RHEL/CentOS 8 - x86_64       1.2 MB/s | 576 kB     00:00    
PostgreSQL 9.5 for RHEL/CentOS 8 - x86_64       384 kB/s | 483 kB     00:01    
PostgreSQL 9.4 for RHEL/CentOS 8 - x86_64       292 kB/s | 415 kB     00:01    
Dependencies resolved.
================================================================================
 Package                Arch        Version                   Repository   Size
================================================================================
Installing:
 postgresql96           x86_64      9.6.16-1PGDG.rhel8        pgdg96      1.5 M
Installing dependencies:
 postgresql96-libs      x86_64      9.6.16-1PGDG.rhel8        pgdg96      347 k

Transaction Summary
================================================================================
Install  2 Packages

Total download size: 1.8 M
Installed size: 10 M
Downloading Packages:
(1/2): postgresql96-libs-9.6.16-1PGDG.rhel8.x86 1.2 MB/s | 347 kB     00:00    
(2/2): postgresql96-9.6.16-1PGDG.rhel8.x86_64.r 4.6 MB/s | 1.5 MB     00:00    
--------------------------------------------------------------------------------
Total                                           5.6 MB/s | 1.8 MB     00:00     
warning: /var/cache/dnf/pgdg96-b6cbc5f1e3bd22e1/packages/postgresql96-9.6.16-1PGDG.rhel8.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 442df0f8: NOKEY
PostgreSQL 9.6 for RHEL/CentOS 8 - x86_64       1.6 MB/s | 1.7 kB     00:00    
Importing GPG key 0x442DF0F8:
 Userid     : "PostgreSQL RPM Building Project <pgsqlrpms-hackers@pgfoundry.org>"
 Fingerprint: 68C9 E2B9 1A37 D136 FE74 D176 1F16 D2E1 442D F0F8
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64            1/2 
  Running scriptlet: postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64            1/2 
  Installing       : postgresql96-9.6.16-1PGDG.rhel8.x86_64                 2/2 
  Running scriptlet: postgresql96-9.6.16-1PGDG.rhel8.x86_64                 2/2 
  Verifying        : postgresql96-9.6.16-1PGDG.rhel8.x86_64                 1/2 
  Verifying        : postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64            2/2 
Installed products updated.

Installed:
  postgresql96-9.6.16-1PGDG.rhel8.x86_64                                        
  postgresql96-libs-9.6.16-1PGDG.rhel8.x86_64                                   

Complete!
```
i forgot to point this out when raised the PR. 
btw where do you look for UBI rpms?

Reply:
OK, if this works then let's proceed. We will be notified from monitoring you configured in the job if anything goes wrong.

> btw where do you look for UBI rpms?

try `dnf whatprovides pg_dump`